noir-projects/aztec-nr/aztec/src/utils/conversion/fields_to_bytes.nr

```diff
@@ -0,0 +1,87 @@
+/// These functions are used to facilitate the conversion of log plaintext represented as fields into bytes and back.
+///
+/// `fields_to_bytes` uses fixed-size arrays since encryption contexts have compile-time size information.
+/// `fields_from_bytes` uses BoundedVec for flexibility in unconstrained contexts where sizes are dynamic.
+///
+/// Together they provide bidirectional conversion between fields and bytes.
+
+/// Converts an input array of fields into a single array of bytes.
+/// Each field is converted to a 32-byte big-endian array.
+/// Since a field is ~254 bits, you'll end up with a subtle 2-bit "gap" at the big end,
+/// every 32 bytes. Be careful that such a gap doesn't leak information!
```

Comment:
I'd expand a little bit here, maybe with a simple example. And what do you mean by leak information?

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/12581/commits/f343f7d6a3f635a36fd4b9b66b6f7132676455f3